chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -424,7 +424,11 @@
     Provisioner::Repositories.inspect_repos(node)
     repos = Provisioner::Repositories.get_repos(node, "suse", target_platform_version)
 
-    packages = node[:provisioner][:packages][os] || []
+    if node[:provisioner][:packages]
+      packages = node[:provisioner][:packages][os] || []
+    else
+      packages = []
+    end
 
```

Comment:
Actually I had your version before the evil rewrite. But I tend to favor my current style just for readability reasons. `If`/`else` seems to reflect the logic better than overwriting a set value in a simple `if`. But I guess this is a matter of taste and I'm happy to change it if someone else outvotes me.